uk.co.compucorp.civicrm.hrcomments/CRM/HRComments/BAO/Comment.php

```diff
@@ -0,0 +1,24 @@
+<?php
+
+class CRM_HRComments_BAO_Comment extends CRM_HRComments_DAO_Comment {
+
+  /**
+   * Create a new Comment based on array-data
+   *
+   * @param array $params key-value pairs
+   * @return CRM_HRComments_DAO_Comment|NULL
+   */
+  public static function create($params) {
+    $className = 'CRM_HRComments_BAO_Comment';
+    $entityName = 'Comment';
+    $hook = empty($params['id']) ? 'create' : 'edit';
+
+    CRM_Utils_Hook::pre($hook, $entityName, CRM_Utils_Array::value('id', $params), $params);
+    $instance = new $className();
```

Comment:
you can replace this with 
```php
$instance = new self();
```

And then you won't need the `$className` variable at all